includes/abstracts/abstract-wc-log-handler.php

```diff
@@ -0,0 +1,135 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+if ( ! class_exists( 'WC_Logger' ) ) {
+	include_once( dirname( dirname( __FILE__ ) ) . '/class-wc-logger.php' );
+}
+
+/**
+ * Abstract WC Log Handler Class
+ *
+ * @version        2.8.0
+ * @package        WooCommerce/Abstracts
+ * @category       Abstract Class
+ * @author         WooThemes
+ */
+abstract class WC_Log_Handler {
+
+	/**
+	 * Minimum log level this handler will process.
+	 *
+	 * @var int 0-8 minimum level severity for handling log entry.
+	 * @access private
+	 */
+	protected $_threshold;
+
+	/**
+	 * Log levels by severity.
+	 *
+	 * @var array
+	 * @access private
+	 */
+	protected static $_log_levels = array(
+		WC_Logger::DEBUG     => 0,
+		WC_Logger::INFO      => 1,
+		WC_Logger::NOTICE    => 2,
+		WC_Logger::WARNING   => 3,
+		WC_Logger::ERROR     => 4,
+		WC_Logger::CRITICAL  => 6,
+		WC_Logger::ALERT     => 7,
+		WC_Logger::EMERGENCY => 8,
+	);
+
+	/**
+	 * Constructor for log handler.
+	 *
+	 * @param arr $args {
+	 *     @type string $threshold Optional. Default 'emergency'. Sets the log severity threshold.
+	 *         emergency|alert|critical|error|warning|notice|info|debug
+	 * }
+	 */
+	public function __construct( $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'threshold' => WC_Logger::EMERGENCY,
```

Comment:
> So you need to be careful if you display a message like "EMERGENCY" or "CRITICAL" by default in from a debug message, since the users can get really concerned about it.

To clarify, log messages have **no default level**, a user *must* provide a level when they `log` a message. The only way around this is by using the deprecated `WC_Logger->add()`, which I believe I set to log an `info` level message.

The portion of code we're discussing here is the default threshold for `WC_Log_Handler`. `WC_Logger` delegates to work of actually producing some log output to implementations of `WC_Log_Handler`. The handler has a threshold which is the minimum log level it will handle. This way a handler can ignore some messages, for example an email handler probably doesn't want to send `debug` messages, so it would have a higher threshold.

This will not affect any *content* of log messages. Users will never see `EMERGENCY [...]` unless an actually emergency level log has been written.